launcher/Application.h

```diff
@@ -44,6 +45,8 @@
 #include <QUrl>
 #include <updater/GoUpdate.h>
 
+#include "ui/themes/ThemeManager.h"
```

Comment:
You can just forward declare instead, as you are just declaring the pointer in this header, see:
https://github.com/PrismLauncher/PrismLauncher/blob/3c571c074f0544a674671dd6e7f45ecf84f0c06b/launcher/Application.h#L54

then you can include this in Application.cpp

Reply:
so I would do this?
```suggestion
class ThemeManager;"
```
in Application.h and move the #include to the .cpp file?

I needed to look up what forward declaring something means and why you could need it.

In this case I suppose it would just mean, that other parts of code, which #include Application.h wouldn't automatically include the entire ThemeManager?

Reply:
yeah, you would put `class ThemeManager;` after https://github.com/PrismLauncher/PrismLauncher/blob/3c571c074f0544a674671dd6e7f45ecf84f0c06b/launcher/Application.h#L73-L74

then move the `#include "ui/themes/ThemeManager.h"` to Application.cpp

> In this case I suppose it would just mean, that other parts of code, which #include Application.h wouldn't automatically include the entire ThemeManager?

yup, so when you touch ThemeManager.h it just recompiles a few files instead of a hundred ^^

Reply:
basically, when declaring a pointer (`A* a;`) or something that doesn't need to know the size of a class like `std::vector<A> v;` you can forward declare A like so `class A;` you need the definition of A when you use it (member access, dereference, ...). when doing forward declaration you avoid including a whole file in a header that would be included in a whole lot of other files, this will avoid recompiling the header of A a lot speeding up recompilation when developing :) but also compilation as you don't compile the header of A too many times.